### PR TITLE
feat(sdk): bool & string tests

### PR DESCRIPTION
**Pull request by revitalbarletz**
## Checklist

- [ ] Title matches [Winglang's style guide](https://docs.winglang.io/contributing/pull_requests#how-are-pull-request-titles-formatted)
- [ ] Description explains motivation and solution
- [ ] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: examples/tests/sdk_tests/std/string.w [hunk 1]
```diff
@@ -47,14 +47,105 @@ test "at()" {
...
+test "uppercase()" {
+  assert("BOOM".uppercase() == "BOOM");
+  assert("BooM".uppercase() == "BOOM");
+  assert("boom".uppercase() == "BOOM");
+}
+
+//-----------------------------------------------------------------------------
 // split()
+
+assert("hello;wing".split(";").at(0) == "hello");
```
Comment (staycoolcall911) [inline examples/tests/sdk_tests/std/string.w:114]: let's test index 1 or 2 also, not just 0

File: examples/tests/sdk_tests/std/string.w [hunk 2]
```diff
@@ -47,14 +47,105 @@ test "at()" {
 }
```
Comment (staycoolcall911) [inline examples/tests/sdk_tests/std/string.w:47]: For testing `.at()` - what happens on non happy path cases (out of bounds, negative value, etc.)?

File: examples/tests/sdk_tests/std/string.w [hunk 3]
```diff
@@ -47,14 +47,105 @@ test "at()" {
...
-// TODO: https://github.com/winglang/wing/issues/2785
-
 // concat()
+
+assert("boom".concat("boom") == "boomboom");
+
+test "concat()" {
+  assert("boom".concat("boom") == "boomboom");
+}
+
```
Comment (staycoolcall911) [inline examples/tests/sdk_tests/std/string.w:57]: Missing tests for `.contains()`

**Review by staycoolcall911 — APPROVED** [review]
Nice job! BTW, did you consider using chatgpt for all these tests?

File: examples/tests/sdk_tests/std/bool.w [hunk 4]
```diff
@@ -3,3 +3,14 @@
...
 assert(t == true);
+
+let f = bool.fromJson(Json.parse("false"));
+assert(f == false);
+
+test "fromJson()" {
+    let t = bool.fromJson(Json.parse("true"));
+    assert(t == true);
+
+    let f = bool.fromJson(Json.parse("false"));
```
Comment (eladb) [inline examples/tests/sdk_tests/std/bool.w:14]: What about failure cases?

File: examples/tests/sdk_tests/std/string.w [hunk 5]
```diff
@@ -40,21 +40,148 @@ test "length" {
 // at()
 
 assert("boom".at(2) == "o");
-//TODO: assertThrows("ERROR", () => { "hey".at(-1); });
+// Negative integers count back from the last string character.
```
Comment (eladb) [inline examples/tests/sdk_tests/std/string.w:43]: Is this documented in the api docs?

File: examples/tests/sdk_tests/std/string.w [hunk 6]
```diff
@@ -40,21 +40,148 @@ test "length" {
...
+
+//-----------------------------------------------------------------------------
+// contains()
+
+assert("hello wing".contains("hello"));
+assert(!"hello wing".contains("Hello"));
+assert("hello wing".contains("w"));
+
+test "contains()" {
+  assert("hello wing".contains("hello"));
```
Comment (eladb) [inline examples/tests/sdk_tests/std/string.w:184]: Failure modes missing in many tests

**Review by eladb — APPROVED** [review]

File: examples/tests/sdk_tests/std/string.w (hunk 5 above)
Comment (revitalbarletz) [reply examples/tests/sdk_tests/std/string.w:43]: It is documented in JS. Do you think we should document this? this is the expected behavior in JS.

**Pull request merged**